File: 3264-final-array-state-after-k-multiplication-operations-i/3264-final-array-state-after-k-multiplication-operations-i.java
class Solution {
    public int[] getFinalState(int[] nums, int k, int multiplier) {
        for(int i = 0;i < k;i++){
            int ind = 0;
            for(int j = 0;j < nums.length;j++){
                if(nums[j] < nums[ind]){
                    ind = j;
                }
            }
            nums[ind] *= multiplier;
        }
        return nums;
    }
}

File: 3264-final-array-state-after-k-multiplication-operations-i/3264-final-array-state-after-k-multiplication-operations-i.py
from sortedcontainers import SortedList

class Solution:
    def getFinalState(self, nums: List[int], k: int, multiplier: int) -> List[int]:
        n = len(nums)        
        if multiplier == 1:
            return nums
        sl = SortedList([(x, i) for i, x in enumerate(nums)])
        while True:
            (x, i) = sl[0]
            sl.remove((x, i))
            sl.add((x * multiplier, i))
            k -= 1
            if k == 0:
                answer = [0] * n
                for (x, i) in sl:
                    answer[i] = x
                return answer